### PR TITLE
use Bytes.EMPTY instead of null to indicate empty keys/values

### LINKED ISSUE #9517: Slot values should never be null
### Description

Replace all slot values and keys to use Bytes.EMPTY to indicate empty values

### Steps to reproduce

See above

### Additional context

_No response_

### Hedera network

other

### Version

v0.43

### Operating system

None

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Fix broken HAPI tests by using Bytes.EMPTY instead of null

**Related issue(s)**:

Fixes #9517

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/StorageAccess.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
  *
  * @param key the key of the access
  * @param value the value read or overwritten
- * @param writtenValue if not null, the overwriting value
+ * @param writtenValue if not Bytes.EMPTY, the overwriting value
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/StorageAccess.java:33]: but also the `@Nullable` below and be sure to add the `requiresNonNull` check!

File: hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/api/TokenServiceApi.java [hunk 2]
```diff
@@ -140,10 +140,10 @@ void assertValidStakingElection(
      * Updates the storage metadata for the given contract.
      *
      * @param accountId the id of the contract
-     * @param firstKey       the first key in the storage linked list, null if the storage is empty
+     * @param firstKey       the first key in the storage linked list, empty if the storage is empty
```
Comment (david-bakin-sl) [inline hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/api/TokenServiceApi.java:143]: possibly prefer explicit `Bytes.EMPTY` to "empty"

**Review by david-bakin-sl — CHANGES_REQUESTED** [review]
Initial look

**Review by tinker-michaelj — APPROVED** [review]
LGTM, ty @lukelee-sl !

**Review by david-bakin-sl — APPROVED** [review]
Leaving comments to be addressed in (immediately) subsequent PR in order to unblock build

**Pull request merged**